### ISSUE TITLE
Choosing categories on Newsletter signup

### ISSUE THREAD

**Issue by shtefcs** [TelescopeJS/Telescope#1218]
## The problem

@SachaG  I know that @jelenajjo showed you on **Slack chat**, the theme we are working on, which you can see here http://tutorialsgarden.com. 

As you know yourself, newsletter feature is very important when its come to growing your **userbase** and returning visitors on the site. 

For example  **sidebar.io** or **crater.io** target specific users interested in **design** and **meteor**.

I came to situation where my website is not focused on one specific niche.

**Tutorials Garden** is a community for all kind tutorials related to 2d, 3d graphics, coding, generally Computer & Internet technologies.

## To come straight on the point.

There is need for an option to choose which categories you want to receive in newsletter.

![](http://content.screencast.com/users/shtefcs/folders/Jing/media/03815c2f-b666-495a-8dc9-7191612aa9dd/2015-12-27_1658.png)

It can be simple **check boxes** or **drop down** with **multi select** option.

This way you will not annoy users with unrelated emails which can lead to **unsubscribe** or **spam**, but will fill their inbox with related content they might be interested.

### For example:

As **web designer**, I don't wanna receive posts from **3ds max** or **PHP** categories.
Or as  **javascript** developer, I don't wanna receive updates/tutorials related to **PHP**.

## Conclusion 
It would much better experience and value for both **users** and **webmasters** to have this option.

## Solution
The news letter system should work same as now. Just instead getting content from whole site and send it one newsletter, create newsletter campaign for each category and pull the top/best content.

What do you think guys ?
@queso  @jshimko @rpullinger @splendido 

Peace!

**Comment by jelenajjo** [comment]
:+1:

**Comment by SachaG** [comment]
That's a great idea, I'll accept a PR for it :)

**Comment by SachaG** [comment]
By the way, as you work on this feature keep in mind that it would require a complete change of how the newsletter is managed in Telescope. Currently the newsletter is entirely handled by MailChimp. So this would require either generating different MailChimp lists and newsletters for each category, or else bypassing MailChimp altogether.

**Comment by shtefcs** [comment]
Ok @SachaG , if we do it our self, will send PR .

Btw, can you add me on slack chat ? 

Email: shtefistefi@yahoo.co.uk or stefan@vanila.io

Tnx

**Comment by raduchiriac** [comment]
I know @sungwoncho did something similar on a project we worked on. 
Maybe he can give us more details.

**Comment by SachaG** [comment]
@shtefcs you can sign up for the Slack chat here: http://slack.telescopeapp.org :)

**Comment by sungwoncho** [comment]
@raduchiriac and I previously worked on a project that implemented this feature. We maintained a separate collection to keep `userId` and all `_id`s of subscribed categories. As @SachaG mentioned, you either have to generate a new MailChimp list or use something more suitable for transactional emails. We used Mandrill to send the emails.

Perhaps we could also make a package for this. Do you think it is doable @SachaG?

**Comment by shtefcs** [comment]
@sungwoncho Would you like to share package ?

**Comment by raduchiriac** [comment]
I don't know how much of the code you created is confidential (maybe most of it) but I would be happy to start a new package from the ground-up

**Comment by shtefcs** [comment]
@raduchiriac That would be super nice :+1:

**Comment by cheeseweasel** [comment]
I am going to implement this in this fork: https://github.com/cheeseweasel/Telescope just familiarising myself with telescope at the moment.

If anyone has any thoughts or suggestions, feel free to chime in. :)

**Comment by shtefcs** [comment]
@cheeseweasel How to contact you :) ?

**Comment by raduchiriac** [comment]
I think it's wiser to start working on a plugin. I already did that, we can maybe join forces there

**Comment by cheeseweasel** [comment]
@raduchiriac I'm not sure - it seems like a logical extension to the current newsletter newsletter package - for example they can share templates easily if they're in the same package. Why do you think a separate package makes more sense?

@shtefcs I added my email to my profile. :)

**Comment by cheeseweasel** [comment]
Okay I've been thinking about this (I will get started on it soon :) ), and I wanted to see if people had any opinions on how it should work.

We have a module called `subscribe-to-posts`, although this appears to have been built in a more general fashion so it should be easy to use it to subscribe to other objects too, I'm thinking categories and authors here.

To my mind it makes sense to use that rather than duplicate the functionality elsewhere. If we did that it would be cleaner to rename the module `subscriptions`, but that would obviously be a breaking change, so I'm not sure what you'd want to do here.

**Comment by shtefcs** [comment]
@cheeseweasel I would like to help with that, but I can't. Lets see if @SachaG can just point you, even I know he is over loaded with work.

**Comment by SachaG** [comment]
I'd rather not rename the module because it would require people to manually change the name in their `packages` file. I think it's fine to keep the same name for now, or else create a new package altogether that people can optionally add (and maybe eventually merge subscribe-to-posts into it).

**Comment by cheeseweasel** [comment]
I agree, after thinking about this I had come to the same two options as I certainly wouldn't want to break any current installs.

I will probably create a new package that can exist alongside subscribe-to-posts for the time being, and we can maybe deprecate one later on down the line, obviously giving people with existing installs plenty of warning.

**Comment by cheeseweasel** [comment]
Hi all, first of all, apologies about the length of this...

We do have one other complication, but I believe that once this is resolved, we will be there. However it is a bit complicated.

We have two separate cases here (registered and unregistered users), and both work differently. I am storing category subscriptions on a user, so an unregistered user cannot subscribe to individual categories, only the general newsletter.

If a user is registered, then they can subscribe to individual categories.

The general newsletter is still sent via MailChimp.

The category newsletter is sent via the standard email package (in batches, but that's not what I'm worried about at the moment).

Now, the whole point of the email newsletter is to build a list (for the majority of us I'm assuming).

So if somebody is still subscribed to the category newsletter, they still need to be on a mailchimp list, even though they won't receive the standard newsletter from there.

I currently see three possibilities:

1. We have two lists, a newsletter list and a non newsletter list, a user is subscribed to the non newsletter list if they enable category subscription. General newsletter is sent via mailchimp. **Advantages**: general newsletter sent via single API call, **Disadvantages**: complexity
2. We have one list, we don't send any newsletter emails via the list, it is just there for the site manager to use. **Advantages**: relative simplicity, **Disadvantages**: possible server load if the list is big?
3. We don't use mailchimp at all if category newsletters are enabled, after all in option 2 we're only managing list subscription and unsubscription anyway. **Advantages**: simplicity, **Disadvantages**: how does a user mail their list if they want to?

After writing all of this out, option 2 is settling as the most obvious choice for me - both 1. and 3. add complexity for the site manager.

I would still like people's thoughts, maybe there's something obvious that I've missed? Maybe there's a better way of going about all of this?

**Comment by cheeseweasel** [comment]
Okay, and there is a further complication with 2. & 3. - if someone signs up for the newsletter without an account, we will need to store and manage their email internally anyway.

**Comment by SachaG** [comment]
It sounds like maybe we need a separate `Subscriptions` collection. Each subscription could look something like:

```js
{
  createdAt: February 16th, 2016,
  categories: ["xhd7268", "e82hcw", "aj1738sj"],
  frequency: ["Monday", "Wednesday", "Friday"],
  email: "joe@example.com",
  userId: "xj28w7fhskxl" //optional
}
```

If possible, it'd be great to generalize this into a package that also works outside of Telescope and lets users subscribe to specific events. But maybe that's getting a bit too complex…

**Comment by queso** [comment]
gonna ping @ianserlin here because he is doing interesting things with notifications and Hail.io

**Comment by ianserlin** [comment]
Oh hello @queso :)

The way we accomplish the spirit of this is definitely by dividing communication channels into three parts:

* `people`: contact info and notification preferences (this is separate from `users`)
* `messages`: the actual content
* `conversations`: a container for groups of messages with the people who are participating in them

(Hail.io)[hail.io] isn't so much designed with the major email newsletter use-case in mind, it is designed for many-to-many communication between users over multiple "transports" - e.g. email, sms, push notifications, etc. and in-app chat.

In this case, the `Subscriptions` model suggested by Sacha makes a bit more sense because the subscribers to the newsletter don't care about who else is subscribed to the newsletter or need to communicate with them.

**Comment by cheeseweasel** [comment]
Thanks for the feedback guys. A separate `Subscriptions` collection would make sense to allow us to handle registered and non registered users.

A large part of the concern was about how to manage the MailChimp list, but I realise now I can segment users to keep everyone in the same list, but only send the general newsletter to a portion of the list. Making sure unsubscribes work seamlessly across both groups of users is obviously important and a little complicated, but shouldn't be too much of a problem.

**Comment by yusiphy** [comment]
@ cheeseweasel
I am trying to implement the same. But can't get my head round it. I have all the users pre-registered for a general newsletter, i use mosaico for emails. 
But i want them to have an option in the new newsletter to be able to choose particular topics that they want to hear about.
Can you please be kind and guide me in the right direction.
